platforms/esp/32/fastpin_esp32.h

```diff
@@ -24,68 +11,92 @@ template<uint8_t PIN, uint32_t MASK> class _ESPPIN {
-  // inline static void set(register port_t val) __attribute__ ((always_inline)) { gpio_set_level((gpio_num_t)PIN, val); }
+  inline static void lo() __attribute__ ((always_inline)) {
+      if (PIN < 32) GPIO.out_w1tc = MASK;
+      else GPIO.out1_w1tc.val = MASK;
+  }
+
+  inline static void set(register port_t val) __attribute__ ((always_inline)) {
+      if (val) hi();
+      else lo();
+  }
 
   inline static void strobe() __attribute__ ((always_inline)) { toggle(); toggle(); }
 
-  inline static void toggle() __attribute__ ((always_inline)) { if (PIN < OUTPUT_PIN_LIMIT){ _GPB0._GPO = MASK; } }
+  inline static void toggle() __attribute__ ((always_inline)) { if(PIN < 32) { GPIO.out ^= MASK; } else { GPIO.out1.val ^=MASK; } }
 
   inline static void hi(register port_ptr_t port) __attribute__ ((always_inline)) { hi(); }
   inline static void lo(register port_ptr_t port) __attribute__ ((always_inline)) { lo(); }
   inline static void fastset(register port_ptr_t port, register port_t val) __attribute__ ((always_inline)) { *port = val; }
 
-  inline static port_t hival() __attribute__ ((always_inline)) { if (PIN<OUTPUT_PIN_LIMIT) { return GPIO_OUT_REG | MASK;    }}
-  inline static port_t loval() __attribute__ ((always_inline)) { if (PIN<OUTPUT_PIN_LIMIT) { return GPIO_OUT_REG & ~MASK;   }}
-  inline static port_ptr_t port() __attribute__ ((always_inline)) { if(PIN<OUTPUT_PIN_LIMIT) { return &_GPB0._GPO;   }}
-  inline static port_ptr_t sport() __attribute__ ((always_inline)) { if (PIN<OUTPUT_PIN_LIMIT) {return &_GPB0._GPOS; }}
-  inline static port_ptr_t cport() __attribute__ ((always_inline)) { if (PIN<OUTPUT_PIN_LIMIT) {return &_GPB0._GPOC; }}
+  inline static port_t hival() __attribute__ ((always_inline)) { return MASK; }
+  inline static port_t loval() __attribute__ ((always_inline)) { return ~MASK; }
+
+  inline static port_ptr_t port() __attribute__ ((always_inline)) {
+      if (PIN < 32) return &GPIO.out;
+      else return &GPIO.out1.val;
+  }
+
+  inline static port_ptr_t sport() __attribute__ ((always_inline)) { 
+      if (PIN < 32) return &GPIO.out_w1ts;
+      else return &GPIO.out1_w1ts.val;
+  }
+
+  inline static port_ptr_t cport() __attribute__ ((always_inline)) {
+      if (PIN < 32) return &GPIO.out_w1tc;
+      else return &GPIO.out1_w1tc.val;
+  }
+
   inline static port_t mask() __attribute__ ((always_inline)) { return MASK; }
 
   inline static bool isset() __attribute__ ((always_inline)) { return (0x004 & MASK); }
 };
 
-#define _DEFPIN_ESP32(PIN, REAL_PIN) template<> class FastPin<PIN> : public _ESPPIN<REAL_PIN, (1<<(REAL_PIN & 0xFF))> {};
-
+#define _DEFPIN_ESP32(PIN)  template<> class FastPin<PIN> : public _ESPPIN<PIN, ((uint32_t)1 << PIN)> {};
+#define _DEFPIN_32_33_ESP32(PIN) template<> class FastPin<PIN> : public _ESPPIN<PIN, ((uint32_t)1 << (PIN-32))> {};
 
-#ifdef FASTLED_ESP32_RAW_PIN_ORDER
+_DEFPIN_ESP32(0);
+// _DEFPIN_ESP32(1); Using TX causes flashiness when uploading
```

Comment:
don't block the use of these pins -- yes rx/tx are used if you're flashing via USB - but not if you're flashing wirelessly - also, a deployed device will spend far more of it's life running than being flashed.